src/Features/Core/Portable/Completion/Providers/ImportCompletionProvider/ExtensionMethodImportCompletionHelper.ExtensionMethodSymbolComputer.cs

```diff
@@ -34,6 +35,12 @@ private class ExtensionMethodSymbolComputer
             // The value indicates if we can reduce an extension method with this receiver type given receiver type.
             private readonly ConcurrentDictionary<ITypeSymbol, bool> _checkedReceiverTypes;
 
+            private static readonly AsyncBatchingWorkQueue<Project> _workQueue = new(
+                   TimeSpan.FromSeconds(1),
+                   BatchUpdateCacheAsync,
+                   AsynchronousOperationListenerProvider.NullListener,
+                   CancellationToken.None);
```

Comment:
the null-listener and CT.None are not great.  The lack of a listener means that tests can't await the completion work to be done (Which can help for tests getting deterministic behavior).  The lack of a token means that nothing ever stops this work.

I can think of some ways we can possibly make this work, even at teh 'Features' layer.  They're not great, but i think they woudl work.  Let's chat tomorrow :)

Reply:
I have hooked up listener for both work queues, but not sure how to handle cancellation here.

Reply:
@CyrusNajmabadi  ping, curious about your thoughts on cancellation here

Reply:
yeah, the only thought i could have with cancellation would be to import some workspace service that has a default impl that does CT.None.  But it's also is exported at the EditorFeatures alyer where IThreadingContext is available.

Reply:
OK, done